web/src/components/DandiAppBar.vue

```diff
@@ -11,6 +11,51 @@
       <span>You are currently viewing the data portal.
       Click this button to learn more about the DANDI project.</span>
     </v-tooltip>
+    <v-dialog max-width="600px" v-model="regdialog">
+      <template v-slot:activator="{ on }">
+        <v-btn text :disabled="!loggedIn"
+               class="ml-2 white--text" dark v-on="on">Register</v-btn>
+      </template>
+      <v-card>
+        <v-card-title>
+          <span class="headline">Register a new dataset</span>
+        </v-card-title>
+        <v-card-text>
+          <v-container>
+            <v-row>
+              <v-col cols="12">
+                <v-text-field label="Name*"
+                              hint="Provide a title for this dataset"
+                              persistent-hint
+                              :counter="120"
+                              v-model="name"
+                              required></v-text-field>
+              </v-col>
+              <v-col cols="12">
+                <v-textarea label="Description*"
+                              hint="Provide a description for this dataset"
+                              :counter="3000"
+                              persistent-hint
+                              v-model="description"
+                              required></v-textarea>
+              </v-col>
+            </v-row>
+          </v-container>
```

Comment:
No need to use v-row and v-cols here

```suggestion
          <v-text-field
            label="Name*"
            hint="Provide a title for this dataset"
            persistent-hint
            :counter="120"
            v-model="name"
            required
          />
          <v-textarea
            label="Description*"
            hint="Provide a description for this dataset"
            :counter="3000"
            persistent-hint
            v-model="description"
            required
          />
```